File: prueba-tecnica-backend/src/main/java/com/lautaro_cenizo/service/SeguimientoService.java
package com.lautaro_cenizo.service;


import com.lautaro_cenizo.entity.seguimiento.EstadoDetallado;
import com.lautaro_cenizo.entity.tecnico.Tecnico;

public interface SeguimientoService {

    void actualizarEstado(Integer seguimientoId, EstadoDetallado nuevoEstado,
                          String motivo, Tecnico tecnico);

}

File: prueba-tecnica-frontend/src/app/service/seguimiento.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { EstadoDetallado } from '../models/enums/estadoDetallado';
import { Tecnico } from '../models/class/tecnico';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SeguimientoService {

  private baseUrl = 'http://localhost:8080/api/seguimientos';

  constructor(private http: HttpClient) { }

  actualizarEstado(
    seguimientoId: number, 
    nuevoEstado: EstadoDetallado, 
    motivo: string, 
    tecnico: Tecnico
  ): Observable<void> {
    return this.http.put<void>(
      `${this.baseUrl}/${seguimientoId}/actualizar-estado`,
      tecnico,
      { params: { nuevoEstado: nuevoEstado.toString(), motivo } }
    );
  }
}
